core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -65,6 +65,23 @@ struct ChunkSize {
   ChunkSize(int value_) : value(value_) {}
 };
 
+class Cuda;  // forward decl
```

Comment:
Still need to guard for whether KOKKOS_ENABLE_CUDA is defined or not at this time

Reply:
We actually don't, but it's not an error to guard it

Reply:
You introduce a forward declaration and sidestep the whole debate of whether we want that or not

Reply:
I actually didn't see an invitation to that debate. There are two ways of protecting yourself from code that won't compile:

1. Use preprocessor macros
2. Use template instantiations to make sure that any code that needs guarding is only ever evaluated within a template, in which the template argument is a part of any uses of that guarded concept

I'm personally game for either

Reply:
The discussion is in #3838.